payment/src/it/java/com/yas/payment/config/IntegrationTestConfiguration.java

```diff
@@ -0,0 +1,30 @@
+package com.yas.payment.config;
+
+import dasniko.testcontainers.keycloak.KeycloakContainer;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.testcontainers.containers.PostgreSQLContainer;
+
+@TestConfiguration
+public class IntegrationTestConfiguration {
+
+    @Bean(destroyMethod = "stop")
```

Comment:
You need to annotate this bean method with @ServiceConnection so that spring can use this Postgres container as the test database. Otherwise, H2 database will be used. You can check the test logs for more details.